src/wanderer/lib.ts

```diff
@@ -0,0 +1,246 @@
+  const underwater = loc.environment === "underwater";
+  const skiplist = [
+    ...$locations`The Oasis, The Bubblin' Caldera, Barrrney's Barrr, The F'c'le, The Poop Deck, Belowdecks, 8-Bit Realm, Madness Bakery, The Secret Government Laboratory, The Dire Warren`,
+    ...Location.all().filter((l) => l.parent === "Clan Basement"),
+  ];
+  if (!have($item`repaid diaper`) && have($item`Great Wolf's beastly trousers`)) {
+    skiplist.push($location`The Icy Peak`);
+  }
+  const canUnlock = UnlockableZones.some((z) => loc.zone === z.zone && (z.available() || !z.noInv));
+  return !underwater && !skiplist.includes(loc) && (canAdventure(loc) || canUnlock);
+}
+
+function unlock(loc: Location, value: number) {
+  const unlockableZone = UnlockableZones.find((z) => z.zone === loc.zone);
+  if (!unlockableZone) return canAdventure(loc);
+  if (unlockableZone.available()) return true;
+  if (buy(1, unlockableZone.unlocker, value) === 0) return false;
+  return use(unlockableZone.unlocker);
+}
+
+const backupSkiplist = $locations`The Overgrown Lot, The Skeleton Store, The Mansion of Dr. Weirdeaux`;
+function canWanderTypeBackup(location: Location): boolean {
+  return !backupSkiplist.includes(location) && location.combatPercent >= 100;
+}
+
+function canWanderTypeYellowRay(location: Location): boolean {
+  if (location === $location`The Fun-Guy Mansion` && get("funGuyMansionKills", 0) >= 100) {
+    return false;
+  }
+  return canWanderTypeBackup(location);
+}
+
+const wandererSkiplist = $locations`The Batrat and Ratbat Burrow, Guano Junction, The Beanbat Chamber, A-Boo Peak`;
+function canWanderTypeWander(location: Location): boolean {
+  return !wandererSkiplist.includes(location) && location.wanderers;
+}
+
+export function canWander(location: Location, type: DraggableFight): boolean {
+  switch (type) {
+    case "backup":
+      return canWanderTypeBackup(location);
+    case "yellow ray":
+      return canWanderTypeYellowRay(location);
+    case "wanderer":
+      return canWanderTypeWander(location);
+  }
+}
+
+export class WandererTarget {
+  name: string;
+  value: number;
+  location: Location;
+  prepareTurn: () => boolean;
+
+  /**
+   * Process for determining where to put a wanderer to extract additional value from it
+   * @param name name of this wanderer - for documentation/logging purposes
+   * @param location returns the location to adventure to target this; null only if something goes wrong
+   * @param value the expected additional value of putting a single wanderer-fight into the zone for this
+   * @param prepareTurn attempt to set up, spending meat and or items as necessary
+   */
+  constructor(
+    name: string,
+    location: Location,
+    value: number,
+    prepareTurn: () => boolean = () => true
+  ) {
+    this.name = name;
+    this.value = value;
+    this.location = location;
+    this.prepareTurn = prepareTurn;
+  }
+}
+export type WandererFactory = (type: DraggableFight) => WandererTarget[] | undefined;
```

Comment:
`null` is more purposeful than `undefined`, but is there a compelling reason to not just return `[]`?